applications/CoSimulationApplication/python_scripts/data_transfer_operators/kratos_mapping_with_debug.py

```diff
@@ -0,0 +1,379 @@
+            self._mappers[solver_data.inverse_identifier_tuple].InverseMap(solver_data.variable_destination, solver_data.variable_origin, solver_data.mapper_flags)
+            self.__PostProcessVTKPost(solver_data.inverse_identifier_tuple, from_solver_data, to_solver_data, transfer_options)
+        else:
+            # Generate VTK output for debugging
+            self.__GenerateProcessVTK(from_solver_data, to_solver_data, transfer_options)
+
+            # Get the model parts
+            model_part_origin      = self._GetModelPartFromInterfaceData(from_solver_data)
+            model_part_destination = self._GetModelPartFromInterfaceData(to_solver_data)
+
+            if model_part_origin.IsDistributed() or model_part_destination.IsDistributed():
+                mapper_create_fct = python_mapper_factory.CreateMPIMapper
+            else:
+                mapper_create_fct = python_mapper_factory.CreateMapper
+
+            if self.echo_level > 0:
+                info_msg  = "Creating Mapper:\n"
+                info_msg += '    Origin: ModelPart "{}" of solver "{}"\n'.format(solver_data.model_part_origin_name, from_solver_data.solver_name)
+                info_msg += '    Destination: ModelPart "{}" of solver "{}"'.format(solver_data.model_part_destination_name, to_solver_data.solver_name)
+
+                cs_tools.cs_print_info(colors.bold(self._ClassName()), info_msg)
+
+            mapper_creation_start_time = time()
+            self._mappers[solver_data.identifier_tuple] = mapper_create_fct(model_part_origin, model_part_destination, self.settings["mapper_settings"].Clone()) # Clone is necessary because the settings are validated and defaults assigned, which could influence the creation of other mappers
+            self.number_of_mappers += 1
+
+            if self.echo_level > 2:
+                cs_tools.cs_print_info(colors.bold(self._ClassName()), "Creating Mapper took: {0:.{1}f} [s]".format(time()-mapper_creation_start_time,2))
+
+            self.__PostProcessVTKPre(solver_data.identifier_tuple, from_solver_data, to_solver_data, transfer_options)
+            self._mappers[solver_data.identifier_tuple].Map(solver_data.variable_origin, solver_data.variable_destination, solver_data.mapper_flags)
+            self.__PostProcessVTKPost(solver_data.identifier_tuple, from_solver_data, to_solver_data, transfer_options)
+
+    def __DefaultVTKSettings(self):
+        """
+        Create and return a default VTK output settings using KratosMultiphysics (KM) Parameters.
+
+        Args:
+            self: The instance of the class.
+
+        Returns:
+            KM.Parameters: A parameters object initialized with default settings for VTK output.
+        """
+        # Define default settings for VTK output
+        vtk_default_output_parameters = KM.Parameters("""{
+            "Parameters" : {
+                "model_part_name"                    : "",
+                "file_format"                        : "ascii",
```

Comment:
why ascii?

Reply:
Because this is purely for debugging

Reply:
Yes but you want to debug the mapper not the vtk output 

I am not aware of any issues with binary, snd the size difference is considerable, thus I strongly recommend it